### PR TITLE
Views

### PR DESCRIPTION
**Pull request by natymaty**

### CODE REVIEW

File: lib/models/product.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    private
+    def set_id
+      @@id +=1
+    end
+    def set_price(price)
+      raise ArgumentError unless price.is_a?(Numeric)
+      price
+    end
+  end
+end
```
Comment (Kacper3331) [inline lib/models/product.rb:26]: Add empty lines to separate code.

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    get "/basket" do
+      basket = FetchBasket.new.call
+      erb :"basket/show", locals: { basket: basket }
+    end
+
+    post "/basket" do
+      p params
+      AddItemToBasket.new(params).call
+      redirect "/"
+    end
```
Comment (Kacper3331) [inline app.rb:31]: Remove `p params`

File: Gemfile [hunk 3]
```diff
@@ -1,3 +1,6 @@
 source "https://rubygems.org"
 group :test do
   gem "rspec"
+end
```
Comment (smt116) [inline Gemfile:4]: `test` group should be after `gem "sinatra"` with empty line between.

File: Public/application.css [hunk 4]
```diff
@@ -0,0 +1,3 @@
+body {
+  background-image: url("http://sf.co.ua/14/06/wallpaper-1906165.jpg");
```
Comment (smt116) [inline Public/application.css:2]: what if someone will remove this file?

File: app.rb [hunk 5]
```diff
@@ -0,0 +1,33 @@
+require "bundler/setup"
+require "sinatra/base"
+
+Dir["./lib/**/*.rb"].each{|file| require file}
```
Comment (smt116) [inline app.rb:4]: missing spacebars around curly brackets.

File: app.rb [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+module Shop
+  PRODUCTS = [
+      Product.new( name: "lolz", price: 12 ),
```
Comment (smt116) [inline app.rb:8]: spacebars aronund brackets are not needed here (use them for `{}` but not for `()`)

File: lib/models/product.rb [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+    attr_reader :name, :price, :id
+
+    def initialize(name: , price: )
+      @name = name
+      @price = set_price(price)
+      @id = set_id
+    end
+
+    def price_brutto
+      p price * 1.23
```
Comment (smt116) [inline lib/models/product.rb:13]: Why you have `p` here? What will happen if you use `Shop::Product#price_brutto` in some calculations? will it work as expected?

File: lib/services/add_item_to_basket.rb [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+
+    def call
+      BASKET << Item.new(
+        product_id: @product_id,
+        quantity: @quantity,
+      )
+    end
+  end
+end
+
```
Comment (smt116) [inline lib/services/add_item_to_basket.rb:16]: this empty line is not needed

**Pull request merged**